vllm/v1/engine/detokenizer.py

```diff
@@ -127,6 +139,9 @@ def update(self, new_token_ids: list[int],
 
         # 2) Evaluate stop strings.
         stop_string = None
+        if self.stop and self.reasoning_parser and \
+            not self.reasoning_parser.is_reasoning_end(self.token_ids):
+            return stop_string
         if self.stop and len(self.output_token_ids) > self.min_tokens:
```

Comment:
Move check inside the other if below?
```suggestion
        if self.stop and len(self.output_token_ids) > self.min_tokens:
            if (rp := self.reasoning_parser) and not rp.is_reasoning_end(self.token_ids):
                return None
```